jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoost.scala

```diff
@@ -160,7 +160,7 @@ object XGBoost extends Serializable {
         .map(_.toString.toInt).getOrElse(0)
       val overridedParams = if (numEarlyStoppingRounds > 0 &&
           !params.contains("maximize_evaluation_metrics")) {
-        if params.contains("custom_eval") {
+        if (params.contains("custom_eval")) {
```

Comment:
when `num_early_stopping_rounds` is non-zero, `eval_metric` is a built-in metric,  the `custom_eval` is still in params, just the value is null, so it will still throw an exception. 
So the if condition should be `if (params.contains("custom_eval") && params("custom_eval") != null)` or remove `custom_eval` from params.

Reply:
Hmmm....I think I missed some part of this PR

The reality is early stopping doesn’t work with custom metrics